crates/xilem_web/src/context.rs

```diff
@@ -145,26 +187,77 @@ impl Cx {
         &mut self,
         ns: &str,
         name: &str,
-    ) -> (web_sys::Element, VecMap<CowStr, AttributeValue>) {
+    ) -> (
+        web_sys::Element,
+        VecMap<CowStr, AttributeValue>,
+        VecMap<CowStr, ()>,
+        VecMap<CowStr, CowStr>,
+    ) {
         let el = self
             .document
             .create_element_ns(Some(ns), name)
             .expect("could not create element");
         let attributes = self.apply_attributes(&el);
-        (el, attributes)
+        let classes = self.apply_classes(&el);
+        let styles = self.apply_styles(&el);
+        (el, attributes, classes, styles)
     }
 
     pub(crate) fn rebuild_element(
         &mut self,
         element: &web_sys::Element,
         attributes: &mut VecMap<CowStr, AttributeValue>,
+        classes: &mut VecMap<CowStr, ()>,
+        styles: &mut VecMap<CowStr, CowStr>,
     ) -> ChangeFlags {
         self.apply_attribute_changes(element, attributes)
+            | self.apply_class_changes(element, classes)
+            | self.apply_style_changes(element, styles)
     }
 
     // TODO Not sure how multiple attribute definitions with the same name should be handled (e.g. `e.attr("class", "a").attr("class", "b")`)
     // Currently the outer most (in the example above "b") defines the attribute (when it isn't `None`, in that case the inner attr defines the value)
     pub(crate) fn add_attr_to_element(&mut self, name: &CowStr, value: &Option<AttributeValue>) {
+        // Special-case class so it works with the `class` method
```

Comment:
Interesting idea, I'm not sure whether we should specially handle attributes that are given here, since that would apply to potentially a *lot* of attributes (when they're implemented somewhat in the future). Such as `circle().fill()` (which would bloat the wasm binary size).

Maybe we should explicitly tell the user, that there's either the possibility via `attr` (which would then be discouraged) or typed methods (such as `el.class()`). And that using both at the same time will result in undefined behavior or overwrites previous set values. But I'm not sure about that. We could also introduce a warning (in debug builds?) or even an error when `attr("class")` is used.